types/address/hash_test.go

```diff
@@ -75,6 +76,21 @@ func (suite *AddressSuite) TestModule() {
 	assert.NotEqual(addr2, addr3, "changing key must change address")
 }
 
+func unsafeConvertABC() []byte {
+	return unsafeStrToByteArray("abc")
+}
+
+func (suite *AddressSuite) TestUnsafeStrToBytes() {
+	// we convert in other function to trigger GC
+	for i := 0; i < 5; i++ {
+		b := unsafeConvertABC()
+		<-time.NewTimer(5 * time.Millisecond).C
+		b2 := append(b, 'd')
+		suite.Equal("abc", string(b))
+		suite.Equal("abcd", string(b2))
```

Comment:
What's the reasoning behind this test? An append creates a new slice so this doesn't do anything to test out the change, except that append does what it is supposed to do in the standard library. The real test here would be to try to modify the byte slice returned, and you'll see a segmentation fault because that's modified memory that was supposed to be immutable.

Reply:
The reason for this test is to ensure that GC will not eat the underlying array. I added explicit GC call (trigger).

Reply:
We can't make a test for altering bytes because it will lead to segfault - and this is not recoverable.

Reply:
> you'll see a segmentation fault because that's modified memory that was supposed to be immutable

@odeke-em any suggestions on how we should write such a test?